scripts/data-retention-remove-user-data.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+"""
+Our data retention policy is that users have accounts deactivated and personal data stripped after 3 years without
+a login.
+
+This script is very simple and has not been upgraded to accept any arguments to prevent the possibility of accidental
+deletion. If you are in doubt use the dry run option.
+
+Usage: data-retention-remove-user-data.py <stage> <api_token> [--dry-run] [--verbose]
+
+Options:
+    --stage=<stage>                                       Stage to target
+    --api_token=<api_token>                               API Token
+
+    --dry-run                                             List account that would have data stripped
+    --verbose
+    -h, --help                                            Show this screen
+
+Examples:
+    ./scripts/data-retention-remove-user-data.py preview myToken
+    ./scripts/data-retention-remove-user-data.py preview myToken --dry-run --verbose
+
+"""
+import logging
+import sys
+from datetime import datetime, timedelta
+from docopt import docopt
+
+from dmapiclient import DataAPIClient
+
+from dmscripts.helpers.env_helpers import get_api_endpoint_from_stage
+from dmscripts.helpers import logging_helpers
+
+sys.path.insert(0, '.')
+
+
+if __name__ == "__main__":
+    arguments = docopt(__doc__)
+
+    # Get script arguments
+    stage = arguments['<stage>']
+    api_token = arguments['<api_token>']
+
+    dry_run = arguments['--dry-run']
+    verbose = arguments['--verbose']
+
+    # Set defaults, instantiate clients
+    prefix = '[DRY RUN]: ' if dry_run else ''
+    logger = logging_helpers.configure_logger(
+        {"dmapiclient": logging.INFO} if verbose else {"dmapiclient": logging.WARN}
+    )
+    cutoff_date = datetime.now() - timedelta(days=365 * 3)
+    data_api_client = DataAPIClient(base_url=get_api_endpoint_from_stage(stage), auth_token=api_token)
+
+    all_users = list(data_api_client.find_users_iter())
+
+    for user in all_users:
+        last_logged_in_at = datetime.strptime(user['loggedInAt'], '%Y-%m-%dT%H:%M:%S.%fZ')
+        if last_logged_in_at < cutoff_date:
+            logger.info(
+                f"{prefix}Removing personal data for user: {user['id']} with email {user['emailAddress']}"
```

Comment:
Don't we normally hash emails before logging them? Let's log this user that we want to forget 😹 Or is this so we have a fighting chance of reinstating the user if this script goes wrong?

Reply:
Done 👍 Removed this, was over cautious